src/helpers/deserializer.ts

```diff
@@ -1,29 +1,74 @@
 import { AxiosResponse } from 'axios';
+import isArray from 'lodash/isArray';
+import isEmpty from 'lodash/isEmpty';
 
 import { Resource } from 'types/resource';
 
 import { JSONObject } from './json';
 
-export type DeserializableResponse = AxiosResponse<Deserializer>;
+export interface DeserializableResponse extends AxiosResponse<Deserializer> {
+  included?: Deserializer[];
+}
 
-export type DeserializableListResponse = AxiosResponse<Deserializer[]>;
+export interface DeserializableListResponse extends AxiosResponse<Deserializer[]> {
+  included?: Deserializer[];
+}
 
 export interface Deserializer {
   type: string;
   id: string;
   attributes: JSONObject;
+  relationships?: Relationship;
+}
+
+export interface Relationship {
+  [key: string]: {
+    data: RelationshipResource | RelationshipResource[];
+  };
+}
+
+interface RelationshipResource {
+  type: string;
+  id: string;
 }
 
-export const deserialize = <T extends Resource>(data: Deserializer): T => {
+export const deserialize = <T extends Resource>(data: Deserializer, included?: Deserializer[]): T => {
+  const relationships = data.relationships;
   const resource = {
     id: data.id,
     resourceType: data.type,
     ...data.attributes,
   };
 
+  if (relationships && included) {
+    const relationshipResource = (resourceId: string) => {
+      const matchingItem = included.find((item) => item.id === resourceId);
+
+      if (matchingItem) {
+        return deserialize(matchingItem, included);
+      }
+    };
+
+    Object.keys(relationships).forEach((resourceType: string) => {
+      const relationshipData = relationships[resourceType].data;
+
+      if (isArray(relationshipData) && !isEmpty(relationshipData)) {
```

Comment:
```suggestion
      if (!isEmpty(relationshipData || [])) {
```

Reply:
@nvminhtue I want to explicitly check if the data type is an array to distinguish it from the other conditions. 😄
![image](https://github.com/manh-t/react-survey/assets/60863885/ce9c8912-746f-4fb1-8531-0648009e344a)